src/logic/map_objects/tribes/soldier.cc

```diff
@@ -1715,24 +1746,43 @@ void Soldier::naval_invasion_update(Game& game, State& state) {
 		}
 	}
 
-	if (get_position() == state.coords) {
-		if (current_health_ < get_max_health()) {  // Stationed soldiers heal very slowly.
+	if (map.calc_distance(get_position(), state.coords) <= kPortSpaceRadius) {
+		// The target should be unguarded now, conquer the port space.
+		MapRegion<Area<FCoords>> mr(map, Area<FCoords>(portspace_fcoords, kPortSpaceRadius));
+		do {
+			if (mr.location().field->get_owned_by() != owner().player_number()) {
+				molog(game.get_gametime(), "[naval_invasion] Conquering port area\n");
+				game.conquer_area_no_building(PlayerArea<Area<FCoords>>(
+				   owner().player_number(), Area<FCoords>(portspace_fcoords, kPortSpaceRadius)));
+				return start_task_idle(game, descr().get_animation("idle", this), 1000);
+			}
+		} while (mr.advance(map));
+
+		// Stationed soldiers heal very slowly.
+		if (current_health_ < get_max_health()) {
 			constexpr unsigned kStationaryHealPerSecond = 10;
 			heal(kStationaryHealPerSecond);
 		}
+
 		return start_task_idle(game, descr().get_animation("idle", this), 1000);
 	}
 
-	if (start_task_movepath(
-	       game, state.coords, 0, descr().get_right_walk_anims(does_carry_ware(), this), false, 3)) {
+	if (start_task_movepath(game, state.coords, 0,
+	                        descr().get_right_walk_anims(does_carry_ware(), this), false,
+	                        kPortSpaceRadius)) {
 		molog(game.get_gametime(), "[naval_invasion] Return to camp\n");
 		return;
 	}
 
-	if (fcoords.field->get_immovable() != nullptr &&
-	    !fcoords.field->get_immovable()->get_passable()) {
-		// Wait until the fire burns out
-		return schedule_act(game, Duration(100));
+	// Camp is blocked, try to get as close as we can
+	for (int i = 10; i > 0; --i) {
+		Coords coords = game.random_location(state.coords, kPortSpaceRadius);
```

Comment:
(I hope I got this right… :)

`kPortSpaceRadius` is only 2 (and shouldn't be increased), but 30 soldiers can attack from 1 ship, so they may not fit. (tested) How about increasing the search radius by one on the 5th try and maybe once more on the 8th try?

Or explain in a comment and later in the tutorial if limiting the number of invasion soldiers is intended to make it harder to overpower the defendant.